woocommerce/woocommerce-ios
# [REST API] Migrate product variations and attributes

Part of: #8715 

## Description
This PR migrates the product variations and attributes endpoints and updates the related mappers to parse contents without the data envelope.

I apologize for the 500+ changes.
- Majority of those are from mock JSON files.
- I had to club attributes and variations changes together into a single PR to test the changes. (It is not straightforward to fully test variations and attributes separately)

## Testing instructions
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app or skip onboarding if needed.
- On the prologue screen, select Enter your site address and proceed with the address of a self-hosted store.
- Log in with your site credentials. After the login succeeds, you should be navigated to the home screen.
- Navigate to the Products tab and add a "Variable product".
- Tap on "Add Variations" and try adding Attributes
- After adding attributes, try generating variations using the attributes.
- You should be able to create attributes and variations as before.

## Screenshots

| Attributes | Variations |
| - | - |
| <img src="https://user-images.githubusercontent.com/524475/214220142-0ff0f3f0-4d62-4d19-a107-4668a5766c51.png" width="320"> | <img src="https://user-images.githubusercontent.com/524475/214220158-0e155905-0174-4592-b0b1-d52a3617e150.png" width="320"> |

---
